src/main/java/seedu/address/model/tag/ModuleTag.java

```diff
@@ -21,7 +21,7 @@ public class ModuleTag extends Tag implements Comparable<ModuleTag> {
 
     public static final String MESSAGE_CONSTRAINTS =
             "NUS Modules should have 2 - 3 letter prefix, followed by 4 digits and optional 1 - 3 alphabets";
-    public static final String VALIDATION_REGEX = "[A-Z]{2,4}[0-9]{4}[A-Z]{0,3}";
+    public static final String VALIDATION_REGEX = "[a-zA-Z]{2,4}[0-9]{4}[a-zA-Z]{0,3}";
```

Comment:
This is an alternative way of solving it, though I think they achieve the same thing. I'd prefer the latter version to fix #297.